### ISSUE TITLE
Rename virtualisation.* namespace into virtualization.*

### ISSUE THREAD

**Issue by domenkozar** [NixOS/nixpkgs#10723]
Do we even care about british vs american?

**Comment by edolstra** [comment]
No no, we use British spelling :-)

**Comment by domenkozar** [comment]
OK :)

**Comment by michalrus** [comment]
@edolstra, @domenkozar, but not here? :smiley_cat: https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/virtualization

**Comment by lietu** [comment]
You really should reconsider this. Principle of least astonishment applies. I tried multiple times to enable the clearly obvious spelling of `virtualization.vmware.guest.enable = true;`

The entire tech industry has pretty much decided that American English is the one everyone uses. And then you're not even consistent about this? This is *really* an awful anti-feature for Nix right now and pretty much makes it unusable for me.

I can't just waste time all the time banging my head against the wall trying to guess spelling for things because you can't be consistent.

**Comment by lietu** [comment]
The website uses American English:

 - `virtualization`: https://nixos.org/explore.html
 - `stabilization` etc.: https://nixos.org/news.html

Package search finds many results with `virtualization` and none with `virtualisation`:

 - https://search.nixos.org/packages?channel=20.09&from=0&size=30&sort=relevance&query=virtualization
 - https://search.nixos.org/packages?channel=20.09&from=0&size=30&sort=relevance&query=virtualisation

Wiki uses American English:

 - `customization`: https://nixos.wiki/wiki/Keyboard_Layout_Customization
 - `categorized`: https://nixos.wiki/wiki/Nix_channels

Discourse uses American English: `Uncategorized`: https://discourse.nixos.org

The community clearly has settled on using *American* English, and this is an anomaly. You really should simply fix it.

**Comment by domenkozar** [comment]
I agree, since most of the internet uses american English it's better to be consistent.

**Comment by domenkozar** [comment]
@lietu feel free to make a PR, but it should use the renaming module for backwards compatibility.

**Comment by lietu** [comment]
I'm basically trying out NixOS for the first time to see if it's something I could like, so might be a bit much for me atm - I'll consider it if it survives my attempt to pick what to focus on next, got Manjaro and FreeBSD to test out as well 🙂

**Comment by edolstra** [comment]
> The community clearly has settled on using American English, and this is an anomaly. You really should simply fix it.

Disagree. Nix has always used British spelling (e.g. `nix-store --realise`).

**Comment by lietu** [comment]
> Disagree. Nix has always used British spelling (e.g. `nix-store --realise`).

Well then the community is very inconsistent about that and you should just switch everything to American English like pretty much the entire industry has.

I can say that if I have to constantly be guessing which spelling to use for every line I write, I will not be having a good time. I doubt I am alone in this.

**Comment by Gaelan** [comment]
At the bare minimum, the "wrong" spelling should be detected and given a specific error message. That alone would reduce the pain here from a "WTF?" to a "sigh, fine", which is a big improvement.

This is also an issue with the online option search. I was *very* confused the other day when I searched for "specialization" and nothing showed up.

**Comment by toastal** [comment]
Why isn't aliasing and supporting both mentioned? If Erlang can use `Behavior` and  `Behaviour`, and it doesn't seem to matter, why can't Nix?

I run into this whenever I run

```console
# nix optimise store
```

I even out of habit had to backspace my _zee_ (ha) to type that example. 

I'm biased as an American English speaker, although I don't know what would be wrong with just allowing both `-ize` and `-ise` and `-or`, `-our` if it makes people feel more "at home" using Nix. Seeing British ex-coworkers struggle during pair programming with CSS because of `color` vs. `colour` is just needlessly frustrating when both keys could be supported.

**Comment by lietu** [comment]
Aliasing will lead to people writing inconsistent documentation and guides etc. and end-users end up confused as to what is the right way of doing things. Also aliasing tends to cause issues to automatic verification of things in case of e.g. `* { color: red; colour: blue; }` it will be tougher to tell that one of them is completely ignored.

**Comment by toastal** [comment]
Wouldn't it be inconsistent to write documentation in two dialects of English?

I can see where this view comes from a _technical_ viewpoint, but I feel like it causes needless frustration regardless of having one "chosen" from a language and linguistic viewpoint. Being in a country where English is a second language and gap-year international teachers, I speak with frustrated student about having spelling marked wrong each year based arbitrarily on the teacher's country of origin. I understand the compiler doesn't care and it just sees a key it wants satisfied, but it _feels_ like the compiler is telling me I'm using English wrong when it's not. (key word: feels). Does the feeling user's get using the language matter??

>  e.g. `* { color: red; colour: blue; }`

I would expect  `colour: blue` to be chosen because the concept of "alias" would have a special privilege where it literally means the same thing, not just `let color = colour`. It would also prevent `color` being used for one thing and `colour` being used for something different.

**Comment by lietu** [comment]
> I would expect colour: blue to be chosen because the concept of "alias" would have a special privilege where it literally means the same thing, not just let color = colour. It would also prevent color being used for one thing and colour being used for something different.

Exactly, that's what you should expect and it's what I meant. It would be confusing because they are not the same keys, and software seems them as two distinctive keys. Without `colour` defined you can just say it's wrong.

**Comment by lietu** [comment]
>Wouldn't it be inconsistent to write documentation in two dialects of English?

I'm not sure where you think I'm suggesting that, I'm specifically saying that ALIASING would lead to that issue - since both are correct, two different people will choose to write their blog posts etc. using different dialects, end users will see both of them and be confused as to which is correct.

> I understand the compiler doesn't care and it just sees a key it wants satisfied, but it feels like the compiler is telling me I'm using English wrong when it's not. (key word: feels). Does the feeling user's get using the language matter??

Honestly in this regard there's little that can be said except "tough luck", considering most of the software world has settled on American English and that's what you HAVE TO use to get by. It's incredibly rare that something actually forces British English on you like Nix does.

Once we all come up with good ways to solve the aliasing/automatic conversion between the languages then things might be better, but then the issue will not be limited to just `z` vs `s` and `-our` vs `-or` but people will want to write their CSS in French and Chinese as well, which is equally valid/invalid - if you allow one alias then why not multiple, and then we come up with conflicting words in different languages and who gets priority 😄 .. in which case I'd probably go for principle of least astonishment again and just use American English and nothing else.

**Comment by toastal** [comment]
> but people will want to write their CSS in French and Chinese as well, which is equally valid/invalid

This is slipping on a slippery slope at this point :sweat_smile:

 I _do_ still think it worthwhile to cover that "tough luck" where it happens just for comfort. Seeing red squiggly lines in this `<textarea>` in typing examples has got to feel bad to a number of people. It seems this sort of argument comes up in a lot of communities though ... and someone always gets shut out. Usually it's _not_ American English so I shrug, but it's not so fun being in those shoes. :P

**Comment by toastal** [comment]
https://github.com/NixOS/nixpkgs/pull/222236

`fonts.optimizeForVeryHighDPI` has been added. It would be really great if this project could seamlessly support both. Aspell still has its `en_gb-ize` dictionary too.

**Comment by thanegill** [comment]
Worth noting that `pkgs/tools/virtualization` has the American English spelling, so both exist in nixpkgs.

**Comment by quatquatt** [comment]
Would it be possible to add a custom error message, saying something like "Maybe you meant to write virtualisation?"

**Comment by itpropro** [comment]
This is quite confusing, the whole industry uses virtualization with a z. Can this not be standardized and be virtualization across the whole nix ecosystem?

**Comment by adamcstephens** [comment]
I'd encourage anybody motivated to make this happen, but I'm going to close this stale ticket.

**Comment by adamcstephens** [comment]
Fixed. If those thumbs downing me care enough, I hope they'll do the work. As a volunteer project you can make the change you want to see.
